doc/03_research/03_planning/Implementation.md

```diff
@@ -0,0 +1,135 @@
+
+Input:
+
+* Trajectory points (preplanning)
+
+Output:
+
+* Max. Velocity (Acting)
+
+
+### <u>_Update path_</u>
+
+The update path module receives a command from the decision maker, that the current path is not possible to drive without collision. It uses the lanelet modell from the preplanner and chooses the next possible lanelet and accordingly updates the trajectory. It also tells the velocity profile to update because of the new trajectory.
+
+Input:
+
+* lanelet modell (preplanning)
+* update command (decision making)
+* information about blocked lanelets (decision making, perception)
+
+Output:
+* updated trajectory (acting, decision making)
+* update command (velocity profile)
+
+### <u>_Measure distance_</u>
+
+This module measures the distance to obstacles, especially cars, with the Lidar Sensor. The current distance value is published to the acting side for keeping a safe distance (Adaptive Cruise Control).
+
+Input:
+
+* Lidar Sensor data (perception, sensoring)
+
+Output:
+
+* distance value (acting)
+
+---
+## **Decision Making**
+
+Obstacle detection is based on the sensor data from the perception area. If an obstacle is being recognized, the decision making sends a message to the local path planning where the system then chooses another trajectory/lanelet.
+With the Lanelets Model it is possible to give a prediction for other objects and the vehicle itself, by following the lanelet direction of an object.
+
+The decision making can be implemented with a state machine. For every incoming perception there must be a state defined. The system needs to make good predictions to avoid collisions. The Perception data and the Lanelet modell are the main input for the system.
+
+Input:
+
+* Lanelet data (preplanning, local path planning)
+* perception data (traffic lights situation, pedestrians,...)
+
+Output:
+
+* updated driving status (acting, local path planning)
+* Lanelet data (acting)
+
+
+---
+## Next steps
+
+* Implement Map Manager to convert data into a compatible type for route planning and to extract additional informations (Speed Limits, trafic signs, traffic lights)
+* Implement a commonroad route planner (old projects and Gitlab TUM)
+* Analyze Lanelet plan and be familiar with it (Which information can we additionally receive from the plan?)
+* Enrich Lanelet Modell/Map with additional Informations (additional/parallel Lanes, Speed Limits, trafic signs, traffic lights)
+* Choose the Decision Maker (Evaluate Markov Modell in combination with occupancy grid)
+* Read Lidar Sensor data and calculate distances
```

Comment:
Auch hier besteht Potenzial für Überscheidungen mit Perception

Reply:
Ja in dem möglichen Issue steckt dann auch die Abstimmung mit der Perception mit drin. Werden die Daten von der Perception gepublished oder greift das planning direkt auf die Sensordaten zu.